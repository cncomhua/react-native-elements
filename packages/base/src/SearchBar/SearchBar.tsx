import React from 'react';
import { SearchBarIOS, SearchBarIosProps } from './SearchBar-ios';
import { SearchBarAndroid, SearchBarAndroidProps } from './SearchBar-android';
import { SearchBarDefault, SearchBarDefaultProps } from './SearchBar-default';
import {
  ActivityIndicatorProps,
  StyleProp,
  TextStyle,
  ViewStyle,
  TextInput,
} from 'react-native';
import { IconNode } from '../Icon';
import { Theme } from '../helpers';

const SEARCH_BAR_COMPONENTS = {
  ios: SearchBarIOS,
  android: SearchBarAndroid,
  default: SearchBarDefault,
};

export interface SearchBarBaseProps
  extends React.ComponentPropsWithRef<typeof TextInput> {
  containerStyle?: StyleProp<ViewStyle>;
  inputContainerStyle?: StyleProp<ViewStyle>;
  clearIcon?: IconNode;
  searchIcon?: IconNode;
  inputStyle?: StyleProp<TextStyle>;
  loadingProps?: ActivityIndicatorProps;
  showLoading?: boolean;
  leftIconContainerStyle?: StyleProp<ViewStyle>;
  rightIconContainerStyle?: StyleProp<ViewStyle>;
  onClear?(): void;
  onFocus?(): void;
  onBlur?(): void;
  onChangeText?(text: string): void;
  onCancel?(): void;
  theme?: Theme;
}

export type SearchBarProps =
  | SearchBarDefaultProps
  | SearchBarAndroidProps
  | SearchBarIosProps;

export class SearchBar extends React.Component<
  SearchBarProps & {
    theme?: Theme;
    platform?: 'default' | 'ios' | 'android';
  }
> {
  searchBar!: SearchBarIOS;
  static defaultProps = {
    platform: 'default' as const,
  };

  focus = () => {
    this.searchBar.focus();
  };

  blur = () => {
    this.searchBar.blur();
  };

  clear = () => {
    this.searchBar.clear();
  };

  cancel = () => {
    this.searchBar?.cancel();
  };

  render() {
    const Component: typeof React.Component =
      SEARCH_BAR_COMPONENTS[this.props.platform] || SearchBarDefault;

    return (
      <Component
        ref={(ref: SearchBarIOS) => {
          this.searchBar = ref;
        }}
        {...this.props}
      />
    );
  }
}
