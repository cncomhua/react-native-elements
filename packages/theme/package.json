{
  "name": "@react-native-elements/theme",
  "version": "1.0.0",
  "description": "",
  "main": "src/index.tsx",
  "react-native": "src/index.tsx",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "peerDependencies": {
    "@react-native-element/base": "*"
  },
  "author": "",
  "license": "ISC"
}
