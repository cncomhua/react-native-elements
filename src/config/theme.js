// For testing purposes only
import colors from './colors';

export default {
  colors,
};
