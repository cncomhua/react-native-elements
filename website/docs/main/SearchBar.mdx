---
id: searchbar
title: SearchBar
slug: /searchbar
---

import Usage from './usage/SearchBar/SearchBar.mdx';

import Props from './props/SearchBar.mdx';

SearchBars are used to search or filter items. Use a SearchBar when the number
of items directly impacts a user's ability to find one of them.

## Usage

<Usage />

## Props

<Props />
