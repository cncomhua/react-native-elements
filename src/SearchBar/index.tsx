import { withTheme } from '../config';
import { SearchBar, SearchBarProps } from './SearchBar';
import { SearchBarAndroidProps } from './SearchBar-android';
import { SearchBarIosProps } from './SearchBar-ios';
import { SearchBarDefaultProps } from './SearchBar-default';

export { SearchBar };
export type {
  SearchBarProps,
  SearchBarAndroidProps,
  SearchBarDefaultProps,
  SearchBarIosProps,
};
/**
 * RNE SearchBar
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/searchbar)
 */
export default withTheme(SearchBar, 'SearchBar');
