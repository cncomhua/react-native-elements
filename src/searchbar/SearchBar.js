import React from 'react';
import PropTypes from 'prop-types';

import IOSSearchBar from './SearchBar-ios';
import AndroidSearchBar from './SearchBar-android';
import DefaultSearchBar from './SearchBar-default';
import { merge, ThemeConsumer } from '../config';

const SEARCHBAR_COMPONENTS = {
  ios: IOSSearchBar,
  android: AndroidSearchBar,
  default: DefaultSearchBar,
};

const SearchBar = props => {
  const SearchBar = SEARCHBAR_COMPONENTS[props.platform] || DefaultSearchBar;
  return <SearchBar ref={ref => (props.searchBarRef = ref)} {...props} />;
};

SearchBar.propTypes = {
  searchBarRef: PropTypes.func,
  platform: PropTypes.oneOf(['default', 'ios', 'android']),
};

SearchBar.defaultProps = {
  platform: 'default',
};

export default class ThemedSearchBar extends React.Component {
  focus = () => {
    this.searchbar.focus();
  };

  blur = () => {
    this.searchbar.blur();
  };

  clear = () => {
    this.searchbar.clear();
  };

  cancel = () => {
    this.searchbar.cancel && this.searchbar.cancel();
  };

  render() {
    return (
      <ThemeConsumer>
        {({ theme }) => (
          <SearchBar
            searchBarRef={ref => (this.searchbar = ref)}
            {...merge({}, theme.SearchBar, this.props)}
            theme={theme}
          />
        )}
      </ThemeConsumer>
    );
  }
}
