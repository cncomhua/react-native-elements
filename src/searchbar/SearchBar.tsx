import React from 'react';
import { withTheme } from '../config';
import IOSSearchBar, { SearchBarIosProps } from './SearchBar-ios';
import AndroidSearchBar, { SearchBarAndroidProps } from './SearchBar-android';
import DefaultSearchBar, { SearchBarDefaultProps } from './SearchBar-default';
import {
  ActivityIndicatorProps,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';
import { IconNode } from '../icons/Icon';
import { Theme, FullTheme } from '../config/theme';
type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };
const SEARCHBAR_COMPONENTS = {
  ios: IOSSearchBar,
  android: AndroidSearchBar,
  default: DefaultSearchBar,
};

export type SearchBarBaseProps = {
  platform: 'default' | 'ios' | 'android';
  containerStyle?: StyleProp<ViewStyle>;
  inputContainerStyle?: StyleProp<ViewStyle>;
  clearIcon?: IconNode;
  searchIcon?: IconNode;
  inputStyle?: StyleProp<TextStyle>;
  loadingProps?: ActivityIndicatorProps;
  showLoading?: boolean;
  leftIconContainerStyle?: StyleProp<ViewStyle>;
  rightIconContainerStyle?: StyleProp<ViewStyle>;
  onClear?(): void;
  onFocus?(): void;
  onBlur?(): void;
  onChangeText?(text: string): void;
  onCancel?(): void;
  theme?: Theme;
  updateTheme: (updates: RecursivePartial<FullTheme>) => void;
  replaceTheme: (updates: RecursivePartial<FullTheme>) => void;
};

export type SearchBarProps = SearchBarBaseProps &
  SearchBarDefaultProps &
  SearchBarAndroidProps &
  SearchBarIosProps;

class SearchBar extends React.Component<SearchBarBaseProps> {
  searchbar!: IOSSearchBar;
  static defaultProps = {
    platform: 'default' as const,
  };

  focus = () => {
    this.searchbar.focus();
  };

  blur = () => {
    this.searchbar.blur();
  };

  clear = () => {
    this.searchbar.clear();
  };

  cancel = () => {
    this.searchbar.cancel && this.searchbar.cancel();
  };

  render() {
    const Component: typeof React.Component =
      SEARCHBAR_COMPONENTS[this.props.platform] || DefaultSearchBar;

    return (
      <Component
        // @ts-ignore
        ref={(ref: IOSSearchBar) => {
          this.searchbar = ref;
        }}
        {...this.props}
      />
    );
  }
}

export { SearchBar };
export default withTheme(SearchBar, 'SearchBar');
