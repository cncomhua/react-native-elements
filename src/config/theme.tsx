import { ViewProps } from 'react-native';
import { SwipeRatingProps } from '../Rating';
import { TapRatingProps } from '../AirbnbRating';
import { AvatarProps, AccessoryProps } from '../Avatar';
import { BadgeProps } from '../Badge';
import { BottomSheetProps } from '../BottomSheet';
import { ButtonProps } from '../Button';
import { ButtonGroupProps } from '../ButtonGroup';
import { CardProps } from '../card/Card';
import { CheckBoxProps } from '../Checkbox';
import { DividerProps } from '../Divider';
import { HeaderProps } from '../Header';
import { IconProps } from '../Icon';
import { ImageProps } from '../Image';
import { InputProps } from '../Input';
import { ListItemProps } from '../ListItem';
import { ListItemAccordionProps } from '../ListItem';
import { OverlayProps } from '../Overlay';
import { PricingCardProps } from '../PricingCard';
import { SearchBarProps } from '../searchbar/SearchBar';
import { SliderProps } from '../Slider';
import { SocialIconProps } from '../SocialIcon';
import { TextProps } from '../Text';
import { TileProps } from '../Tile';
import { TooltipProps } from '../Tooltip';
import { SwitchProps } from '../Switch';
import { TabItemProps, TabProps } from '../tab/Tab';
import { TabViewProps } from '../tab/TabView';
import { FABProps } from '../FAB';
import { SpeedDialProps, SpeedDialActionProps } from '../SpeedDial';
import { LinearProgressProps } from '../LinearProgress';
import { ChipProps } from '../Chip';
import {
  DialogActionsProps,
  DialogLoadingProps,
  DialogProps,
  DialogTitleProps,
} from '../Dialog';

import colors, { Colors } from './colors';

export default {
  colors,
};

type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };

export interface FullTheme {
  Avatar: Partial<AvatarProps>;
  AvatarAccessory: Partial<AccessoryProps>;
  Badge: Partial<BadgeProps>;
  BottomSheet: Partial<BottomSheetProps>;
  Button: Partial<ButtonProps>;
  ButtonGroup: Partial<ButtonGroupProps>;
  Chip: Partial<ChipProps>;
  Card: Partial<CardProps>;
  CardDivider: Partial<DividerProps>;
  CardFeaturedSubtitle: Partial<TextProps>;
  CardFeaturedTitle: Partial<TextProps>;
  CardImage: Partial<ImageProps>;
  CardTitle: Partial<TextProps>;
  CheckBox: Partial<CheckBoxProps>;
  Divider: Partial<DividerProps>;
  Dialog: Partial<DialogProps>;
  DialogActions: Partial<DialogActionsProps>;
  DialogButton: Partial<ButtonProps>;
  DialogLoading: Partial<DialogLoadingProps>;
  DialogTitle: Partial<DialogTitleProps>;
  Header: Partial<HeaderProps>;
  Icon: Partial<IconProps>;
  Image: Partial<ImageProps>;
  Input: Partial<InputProps>;
  ListItem: Partial<ListItemProps>;
  ListItemAccordion: Partial<ListItemAccordionProps>;
  ListItemButtonGroup: Partial<ButtonGroupProps>;
  ListItemCheckBox: Partial<CheckBoxProps>;
  ListItemContent: Partial<ViewProps>;
  ListItemChevron: Partial<IconProps>;
  ListItemInput: Partial<InputProps>;
  ListItemSubtitle: Partial<TextProps>;
  ListItemTitle: Partial<TextProps>;
  Overlay: Partial<OverlayProps>;
  PricingCard: Partial<PricingCardProps>;
  Rating: Partial<SwipeRatingProps>;
  AirbnbRating: Partial<TapRatingProps>;
  SearchBar: Partial<SearchBarProps>;
  Slider: Partial<SliderProps>;
  SocialIcon: Partial<SocialIconProps>;
  Text: Partial<TextProps>;
  Tile: Partial<TileProps>;
  Switch: Partial<SwitchProps>;
  Tooltip: Partial<TooltipProps>;
  colors: RecursivePartial<Colors>;
  Tab: Partial<TabProps>;
  TabItem: Partial<TabItemProps>;
  TabView: Partial<TabViewProps>;
  TabViewItem: Partial<ViewProps>;
  FAB: Partial<FABProps>;
  SpeedDial: Partial<SpeedDialProps>;
  SpeedDialAction: Partial<SpeedDialActionProps>;
  LinearProgress: Partial<LinearProgressProps>;
}

export type Theme<T = {}> = Partial<FullTheme> & T;

export type UpdateTheme = (updates: RecursivePartial<FullTheme>) => void;

export type ReplaceTheme = (updates: RecursivePartial<FullTheme>) => void;
