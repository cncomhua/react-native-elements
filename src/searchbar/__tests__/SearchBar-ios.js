import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../SearchBar-ios';
import { commonTests, commonPlatformTest } from './common';
import toJson from 'enzyme-to-json';

import theme from '../../config/theme';

describe('iOS SearchBar component', () => {
  commonTests(SearchBar);
  commonPlatformTest(SearchBar);

  describe('Props', () => {
    it('showCancel', () => {
      const component = shallow(<SearchBar theme={theme} showCancel />);

      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
