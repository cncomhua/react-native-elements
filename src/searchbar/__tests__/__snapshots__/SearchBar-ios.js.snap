// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iOS SearchBar component should render with a custom Cancel button title 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Annuler"
  />
</View>
`;

exports[`iOS SearchBar component should render with a custom search icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={<View />}
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render with clear icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#7d7d7d"
          name="ios-close-circle"
          onPress={[Function]}
          size={20}
        />
      </View>
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
    value="test"
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render with clear icon with defaultValue 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    defaultValue="test"
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#7d7d7d"
          name="ios-close-circle"
          onPress={[Function]}
          size={20}
        />
      </View>
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render with container style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      Object {
        "backgroundColor": "gray",
        "paddingBottom": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render with input container style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        Object {
          "backgroundColor": "red",
        },
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render with loading 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      Object {
        "height": 70,
      },
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
          style={
            Array [
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render without clear icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        allowFontScaling={false}
        color="#7d7d7d"
        name="ios-search"
        size={20}
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;

exports[`iOS SearchBar component should render without search icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f5f5",
        "flexDirection": "row",
        "paddingBottom": 13,
        "paddingTop": 13,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    containerStyle={
      Array [
        Object {
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderRadius": 9,
          "height": 36,
          "marginLeft": 15,
        },
        Object {
          "marginRight": 15,
          "width": 718,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "flex": 1,
          "marginLeft": 6,
        },
        undefined,
      ]
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={null}
    onChangeText={[Function]}
    onClearText={null}
    onFocus={[Function]}
    placeholderTextColor="#7d7d7d"
    rightIcon={
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;
